design/dsl/request_test.go

```diff
@@ -0,0 +1,190 @@
+package dsl_test
+
+import (
+	"testing"
+
+	"fmt"
+
+	"goa.design/goa.v2/design"
+	. "goa.design/goa.v2/design/dsl"
+	"goa.design/goa.v2/eval"
+)
+
+func TestRequest(t *testing.T) {
+	var RequestType = Type("Request", func() {
+		Description("Optional description")
+		Attribute("required", design.String)
+		Attribute("name", design.String, "a name")
+		Required("required")
+	})
+	//useful for this test but not others so defined here
+	commonRequestTypeAsserts := func(t *testing.T, o design.Object, ut *design.UserTypeExpr) {
+		keys := []string{}
+		for k, _ := range o {
+			keys = append(keys, k)
+		}
+		assertHasAll(t, []string{"required"}, ut.AllRequired())
+		assertHasAll(t, []string{"required", "name"}, keys)
+		assertDescription(t, "", o["required"].Description)
+		assertDescription(t, "a name", o["name"].Description)
+		assertAttributeType(t, o["name"], design.String)
+		assertAttributeType(t, o["required"], design.String)
+	}
+	cases := map[string]struct {
+		DSL    func()
+		Assert func(t *testing.T, req design.DataType)
+	}{
+		"ArrayOfRequest": {
+			DSL: func() {
+				Request(ArrayOf(RequestType))
+			},
+			Assert: func(t *testing.T, req design.DataType) {
+				v, ok := req.(*design.Array)
+				if !ok || v == nil {
+					t.Errorf("expected request to be a design.Array got %s ", req.Name())
+					return
+				}
+				ut, err := populateAttr(v.ElemType)
+				if err != nil {
```

Comment:
I would pass the `testing.T` to `populateAttr` and have it fail the test rather than test for error here.